### ISSUE TITLE
Invalid reference if related branch does not exist locally

### ISSUE THREAD

**Issue by socketbox** [despreston/gh-worktree#4]
Thanks for doing this work. It's been a real boon to my worktree-based existence.

This error had me scratching my head for perhaps longer than it should have, but if the user has not fetched the branch a PR is based on, a somewhat cryptic error is returned.
```                       
fatal: invalid reference: Branch-name-of-branch-not-yet-fetched   
Error: exit status 128                                            
Usage:                                                            
  worktree pr [pr number] <path> [flags]                          
                                                                  
Flags:                                                            
  -h, --help   help for pr                                        
                                                                  
exit status 128                                                   
```
Something a little more informative might be helpful like, "Branch does not exist locally. Fetch the branch this PR is based upon."

**Comment by despreston** [comment]
I appreciate the kind words. I'm happy to take a look at this when I can, and thank you very much for reporting it.

**Comment by despreston** [comment]
@socketbox I've looked into this briefly. This particular error is from Github. That said, I'm not opposed to making this UX a little better. Would you find it beneficial to know when an error is coming from Github vs something from the gh-worktree plugin?

**Comment by zmre** [comment]
I'm new to this extension, but hitting the same problem. `gh worktree pr 1111 ./pr-1111` gives me something like `fatal: invalid reference: dependabot/something/something`.  I tried doing a `git fetch` and also a `git pull` and the same thing keeps happening.  I tried `git fetch origin pull/1111/head` too, which fetches the pull branch, but the `gh-worktree` command still fails.

In the end, I abandoned the extension and just did this:

```bash
git fetch origin pull/1111/head:pr-1111
git worktree add pr-1111
```

But I thought `gh-worktree` was supposed to do this for me.  Clearly there's something I'm not understanding. Ideally, gh-worktree would automatically do all the steps including any required fetching, but barring that, ideally it would spell out what needs to be done.  Also, I haven't yet tried to make sure I can push to this branch, but I think that using the `pull` namespace made it read only.  So what I think I should have done was to resolve the PR back to a branch and then done the above steps with that branch instead.

But again, what is this plugin doing if not that?

**Comment by despreston** [comment]
Hi @zmre, what's happening under the hood is just:

- get the current repo
- fetching the details about the PR and grabbing the branch name from that
- calling `git worktree add` with the path and the branch name

Im not 100% certain but it's worth remembering that `git fetch` doesnt actually setup a local tracking branch for your remote branches. I believe you need to checkout the branch before creating a worktree from that branch.

**Comment by zmre** [comment]
Appreciate what you've done here and your answer. In context of checking out a PR (presumably someone else's or you'd already have a worktree) it seems the fetch or checkout would always be missing, right? But I wouldn't know what to checkout without the API call. It seems to me that this tool will always fail, right? At least on first call before running other necessary commands?  Or maybe there's some git preference at play here that I don't have setup properly. 

In short, I'm wondering if the fetch or checkout step should just always be done automatically as part of this extension. It seems this will often fail otherwise. 

I'm going to experiment more, but it seems to me right now that a more reliable workflow is actually to add a worktree for master/main into a folder called `pr-XXX` or whatever, `cd` into it, then do a `gh pr checkout XXX`. This method works even when the PR is based off of a fork or when the branch is from dependabot. I'm not sure if this extension should try to mimic that or take another approach but right now it is a coin flip for me if this will work.

**Comment by zmre** [comment]
Decided to make a [proof of concept](https://github.com/zmre/gh-worktree) as a dead simple bash script.  I tried it across the various PRs where this gh extension was failing me and this approach has worked consistently.  At this point, I'm not publishing it so it won't compete with this extension, but where this one gives mysterious errors and leaves the user to try to figure out why it doesn't work (like in dependabot branches, pull requests from forks, etc.), the experiment I put together just works across these because it uses an approach that relies on `gh pr`, which already works for getting these branches and doing whatever git accounting is required to make them work properly.

**Comment by despreston** [comment]
@zmre When I have some time I can take a look at incorporating your idea into this extension. Ive been rather busy over the last year.

**Comment by zmre** [comment]
Yep no worries. I understand. Just some suggestions.  If I were a Go dev, I would've made a PR for you.
